packages/data-broker/src/services/dhis/DhisInputSchemeResolvingApi.js

```diff
@@ -0,0 +1,306 @@
+      // the endpoint used /api/analytics/rawData.json only allows a single "inputIdScheme", which means
+      // both the dataElements and orgUnits need to be ids
+
+      modifiedQuery = await this.replaceDataElementCodesWithIds(modifiedQuery);
+      modifiedQuery = await this.replaceOrgUnitCodesWithIds(modifiedQuery);
+
+      modifiedQuery.inputIdScheme = 'uid';
+    }
+
+    // console.log('DhisInputSchemeResolvingApi', 'getAnalytics', 'modifiedQuery', modifiedQuery);
+    return this.api.getAnalytics(modifiedQuery);
+  }
+
+  async getEventAnalytics(query) {
+    // console.log('DhisInputSchemeResolvingApi', 'getEventAnalytics', 'query', query);
+    let modifiedQuery = { ...query };
+
+    const originalDataElementIdScheme = query.dataElementIdScheme;
+
+    const areDataElementsUsingDhisId = await this.anyDataElementsHaveDhisId(query);
+
+    if (areDataElementsUsingDhisId) {
+      modifiedQuery = await this.replaceDataElementCodesWithIds(modifiedQuery);
+      modifiedQuery.dataElementIdScheme = 'id';
+    }
+
+    if (await this.anyProgramsHaveDhisId(query)) {
+      modifiedQuery = await this.replaceProgramCodesWithIds(modifiedQuery);
+    }
+
+    if (await this.anyOrgUnitsHaveDhisId(query)) {
+      modifiedQuery = await this.replaceOrgUnitCodesWithIds(modifiedQuery);
+    }
+
+    // console.log('DhisInputSchemeResolvingApi', 'getEventAnalytics', 'modifiedQuery', modifiedQuery);
+
+    const response = await this.api.getEventAnalytics(modifiedQuery);
+
+    if (areDataElementsUsingDhisId && originalDataElementIdScheme === 'code') {
+      // The api response will contain data elements with ids, and DhisApi will not be able to translate these
+      // back into codes (because the codes are not set in dhis).
+      // So, we have to do it ourselves using the internal mapping
+
+      const dataElementIdToCode = {};
+
+      // const dataElements = await this.models.dataSource.find({ id: query.dataElementCodes });
+      //
+      // for (const dataElement of dataElements) {
+      //   if (dataElement.config.dhisId) {
+      //     dataElementIdToCode[dataElement.dhisId] = dataElement.code;
+      //   }
+      // }
+
+      for (const dataElementCode of query.dataElementCodes) {
+        if (DATA_SOURCE_CONFIG_BY_CODE[dataElementCode].config.dhisId) {
+          dataElementIdToCode[
+            DATA_SOURCE_CONFIG_BY_CODE[dataElementCode].config.dhisId
+          ] = dataElementCode;
+        }
+      }
+
+      const translated = await this.api.translateElementKeysInEventAnalytics(
+        response,
+        dataElementIdToCode,
+      );
+
+      console.log('getEventAnalytics', 'translated', translated);
+
+      return translated;
+    }
+
+    return response;
+  }
+
+  /**
+   * @param query {*}
+   * @returns bool
+   * @private
+   */
+  anyDataElementsHaveDhisId = async query => {
+    const { dataElementCodes } = query;
+
+    // const dataElements = await this.models.dataSource.find({ code: dataElementCodes });
```

Comment:
Should even be able to check for the presence of `config.dhisId` in the query, eventually :-)